### PR TITLE
[ci] fix build upload info for windows

### PR DESCRIPTION
**Pull request by can-anyscale**
Need another en variable for go/flaky upload to work in windows. Currently failing with https://buildkite.com/ray-project/postmerge/builds/2191#018c6eb6-d391-4b82-a804-a9b60aa19e05/5929-5945

Validating with https://buildkite.com/ray-project/postmerge/builds/2199

Test:
- CI

### CODE REVIEW

File: ci/ray_ci/test_windows_container.py [hunk 1]
```diff
@@ -42,6 +42,8 @@ def test_get_run_command() -> None:
         "-i",
         "--rm",
         "--env",
+        "BUILDKITE",
+        "--env",
```
Comment (aslonnie) [inline ci/ray_ci/test_windows_container.py:46]: maybe programmatically generate this `--env` list here based on `_DOCKER_ENV` ?

Comment (can-anyscale) [reply ci/ray_ci/test_windows_container.py:46]: good idea

File: ci/ray_ci/test_windows_container.py [hunk 2]
```diff
@@ -36,23 +37,16 @@ def _mock_subprocess(inputs: List[str], stdout, stderr) -> None:
 
 def test_get_run_command() -> None:
     container = WindowsContainer("test")
+    envs = []
+    for env in _DOCKER_ENV:
```
Comment (aslonnie) [inline ci/ray_ci/test_windows_container.py:41]: does linux have the same thing? why does the linux one do not have this issue?

**Review by aslonnie — APPROVED** [review]

File: ci/ray_ci/test_windows_container.py (hunk 2 above)
Comment (can-anyscale) [reply ci/ray_ci/test_windows_container.py:41]: linux has BUILDKITE=true set inside the base docker image

**Pull request merged**